lib/testScenario/swaggerAnalyzer.ts

```diff
@@ -245,9 +315,90 @@ export function analyzeExampleDependency(example: SwaggerExample): Dependency[]
       const provider = getProvider(it.value);
       if (provider !== undefined) {
         const resourceChain = getResourceFromPath(it.value);
-        ret.push({ resourceProvider: provider, resourceChain: resourceChain });
+        const resourceType = getResourceTypePath(resourceChain, provider);
+        ret.push({
+          resourceProvider: provider,
+          resourceChain: resourceChain,
+          resourceType: resourceType,
+          jsonPointer: it.pointer,
+          jsonPath: it.path,
+        });
       }
     }
   }
   return ret;
 }
+export interface DependencyResult {
+  apiVersion: string;
+  exampleName: string;
+  resourceType: string;
+  resourceProvider: string;
+  dependentResourceType: string;
+  exampleJsonPointer: string;
+  swaggerResourceIdJsonPath: string;
+  swaggerFilePath: string;
+  exampleFilePath: string;
+  operationId: string;
+}
+
+export function normalizeDependency(
+  res: ExampleDependency[],
+  fileRoot = "specification"
+): DependencyResult[] {
+  const dependency: any = {};
+  const vis = new Set<string>();
+  res
+    .filter((it) => it.externalDependency.length > 0)
+    .map((it) => {
+      return {
+        swaggerFilePath: it.swaggerFilePath.substr(
+          it.swaggerFilePath.indexOf(fileRoot),
+          it.swaggerFilePath.length
+        ),
+        ids: it.externalDependency.map((dependency) => {
+          return {
+            apiVersion: it.apiVersion,
+            exampleName: it.exampleName,
+            resourceType: it.resourceType,
+            resourceProvider: it.resourceProvider,
+            dependentResourceType: dependency.resourceType,
+            operationId: it.operationId,
+            exampleJsonPointer: dependency.jsonPointer,
+            swaggerResourceIdJsonPointer: dependency.resourceIdJsonPointer,
+            exampleFilePath: it.exampleFilePath.substr(
+              it.exampleFilePath.indexOf(fileRoot),
+              it.exampleFilePath.length
+            ),
+          };
+        }),
+      };
+    })
+    .forEach((it) => {
+      if (dependency[it.swaggerFilePath] === undefined) {
+        dependency[it.swaggerFilePath] = [];
+      }
+      for (const id of it.ids) {
+        if (!vis.has(uniqueIndex(id.exampleJsonPointer!, it.swaggerFilePath, id.resourceType!))) {
+          dependency[it.swaggerFilePath].push(id);
+          vis.add(uniqueIndex(id.exampleJsonPointer!, it.swaggerFilePath, id.resourceType!));
+        }
+      }
+    });
+  let ret: DependencyResult[] = [];
+  for (const [k, v] of Object.entries(dependency)) {
+    ret = ret.concat(
+      (v as any).map((it: any) => {
+        return { ...it, swaggerFilePath: k };
+      })
+    );
+  }
+  return ret;
+}
+
+const uniqueIndex = (
+  jsonPointer: string,
+  swaggerFilePath: string,
+  resourceType: string
+): string => {
+  return `${jsonPointer}_${swaggerFilePath}_${resourceType}`;
```

Comment:
Should we add repository, branch, and commit info?

Reply:
No, we could add these infomation in our scripts.